### PR TITLE
npc remover

### PR DESCRIPTION
**Pull request by maxstadelmaier**

### CODE REVIEW

**Comment by maxstadelmaier** [comment]
pls pull

File: analyze.py [hunk 1]
```diff
@@ -19,11 +19,13 @@
 from operator import itemgetter
 from datetime import datetime, timedelta
 import matplotlib.pyplot as plt
+import matplotlib
 import argparse
 import functools
 from scipy.signal import correlate
 import numpy
-from pyvis.network import Network
+# from pyvis.network import Network
```
Comment (LeStahL) [inline analyze.py]:
```suggestion
```

File: analyze.py [hunk 2]
```diff
@@ -247,34 +268,52 @@ def contributorTimeline(self, name):
...
             nWords = contributorData[2]
             timeline = contributorData[3]
 
+            x = timeline.keys()
+            y = timeline.values()
+            x = numpy.array(list(x))
+            y = numpy.array(list(y))
+
+            if args.start_date is not None:
+                for i in range(len(x)):
```
Comment (LeStahL) [inline analyze.py]: I'd prefer a `list(filter(lambda key,value: key > args.start_date.date(), timeline))` type construct
and plotting the values of that (optionally minus their first)

less code is easier to read & maintain

File: analyze.py [hunk 3]
```diff
@@ -247,34 +268,52 @@ def contributorTimeline(self, name):
...
 
-            plt.plot_date(timeline.keys(), timeline.values(), '', label=contributor)
+            plt.plot_date(x, y, '', label=contributor, zorder=y[-1])
 
         plt.legend()
-        plt.grid()
-        plt.xlabel("Verstrichene Lebenszeit")
+        plt.grid(linestyle=":", color="silver")
+        plt.xlabel("Lebenszeit")
+        # plt.yscale("log")
```
Comment (LeStahL) [inline analyze.py]:
```suggestion
```

**Review by LeStahL — COMMENTED** [review]
Thanks! I suggest changes below.

**Comment by maxstadelmaier** [comment]
thanks for your fruitful comments! I adjusted the code and made some more improvements

File: analyze.py [hunk 4]
```diff
@@ -166,6 +167,8 @@ def contributorTimeline(self, name):
 
     if args.start_date is not None:
         args.start_date = datetime.strptime(args.start_date, '%Y-%m-%d')
+    # else:
+    #     args.start_date = datetime(1990,1,1)
```
Comment (LeStahL) [inline analyze.py]:
```suggestion
```

**Comment by LeStahL** [comment]
Thanks!

**Pull request merged**